### ISSUE TITLE
Support c#9 new() target-typed syntax

### ISSUE THREAD

**Issue by angularsen** [angularsen/roslyn-analyzers#8]
This doesn't trigger the analyzer:
```c#
// AssignAll enable
Foo r = new()
{
    Id = ,
    AddedDate = ,
    ModifiedDate = 
};
// AssignAll disable
```

But this does:
```c#
// AssignAll enable
Foo r = new Foo
{
    Id = ,
    AddedDate = ,
    ModifiedDate = 
};
// AssignAll disable
```
